# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1244488336 14400
#      Mon Jun 08 15:12:16 2009 -0400
# Node ID 1a1c839625b33e8040369f3bbe3e38caa2f71137
# Parent  96abf8f7e5b66b3dd8536b4c585233f015cc4d62
fix exist for function handles and inline functions

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-06-08  John W. Eaton  <jwe@octave.org>
+
+	* variables.cc (symbol_exist): Returnn 1 for function handles and
+	inline function objects.
+
 2009-06-08  Jaroslav Hajek  <highegg@gmail.com>
 
 	* symtab.h (symbol_table::do_clear_global,
 	symbol_table::do_clear_global_pattern): Properly erase from both local
 	and global table.
 
 2009-06-08  Jaroslav Hajek  <highegg@gmail.com>
 
diff --git a/src/variables.cc b/src/variables.cc
--- a/src/variables.cc
+++ b/src/variables.cc
@@ -404,17 +404,18 @@ symbol_exist (const std::string& name, c
   if (val.is_defined ())
     {
       bool not_a_struct = struct_elts.empty ();
       bool var_ok = not_a_struct /* || val.is_map_element (struct_elts) */;
 
       if (! retval
 	  && var_ok
 	  && (type == "any" || type == "var")
-	  && (val.is_constant () || val.is_object ()))
+	  && (val.is_constant () || val.is_object ()
+	      || val.is_inline_function () || val.is_function_handle ()))
 	{
 	  retval = 1;
 	}
 
       if (! retval
 	  && (type == "any" || type == "builtin"))
 	{
 	  if (not_a_struct && val.is_builtin_function ())
